util/generate_build_files.py

```diff
@@ -854,49 +269,53 @@ def ExtractVariablesFromCMakeFile(cmakefile):
   return variables
 
 
-def main(platforms):
-  cmake = ExtractVariablesFromCMakeFile(os.path.join('src', 'sources.cmake'))
-  crypto_c_files = (FindCFiles(os.path.join('src', 'crypto'), NoTestsNorFIPSFragments) +
-                    FindCFiles(os.path.join('src', 'third_party', 'fiat'), NoTestsNorFIPSFragments))
-  fips_fragments = FindCFiles(os.path.join('src', 'crypto', 'fipsmodule'), OnlyFIPSFragments)
-  ssl_source_files = FindCFiles(os.path.join('src', 'ssl'), NoTests)
-  tool_c_files = FindCFiles(os.path.join('src', 'tool'), NoTests)
-  tool_h_files = FindHeaderFiles(os.path.join('src', 'tool'), AllFiles)
+def main():
+  cmake = ExtractVariablesFromCMakeFile(os.path.join(SRC_DIR, 'sources.cmake'))
+  crypto_c_files = (FindCFiles(os.path.join(SRC_DIR, 'crypto'), NoTestsNorFIPSFragments) +
+                    FindCFiles(os.path.join(SRC_DIR, 'third_party', 'fiat'), NoTestsNorFIPSFragments))
+  fips_fragments = FindCFiles(os.path.join(SRC_DIR, 'crypto', 'fipsmodule'), OnlyFIPSFragments)
+  ssl_source_files = FindCFiles(os.path.join(SRC_DIR, 'ssl'), NoTests)
+  tool_c_files = FindCFiles(os.path.join(SRC_DIR, 'tool'), NoTests)
+  tool_h_files = FindHeaderFiles(os.path.join(SRC_DIR, 'tool'), AllFiles)
 
   # BCM shared library C files
   bcm_crypto_c_files = [
-      os.path.join('src', 'crypto', 'fipsmodule', 'bcm.c')
+      os.path.join(SRC_DIR, 'crypto', 'fipsmodule', 'bcm.c')
   ]
 
   # Generate err_data.c
-  with open('err_data.c', 'w+') as err_data:
+  if not os.path.isdir(DEST_DIR):
+    os.makedirs(DEST_DIR)
+
+  with open(os.path.join(DEST_DIR, 'err_data.c'), 'w+') as err_data:
     subprocess.check_call(['go', 'run', 'err_data_generate.go'],
-                          cwd=os.path.join('src', 'crypto', 'err'),
+                          cwd=os.path.join(SRC_DIR, 'crypto', 'err'),
                           stdout=err_data)
+
   crypto_c_files.append('err_data.c')
   crypto_c_files.sort()
 
-  test_support_c_files = FindCFiles(os.path.join('src', 'crypto', 'test'),
+  test_support_c_files = FindCFiles(os.path.join(SRC_DIR, 'crypto', 'test'),
                                     NotGTestSupport)
   test_support_h_files = (
-      FindHeaderFiles(os.path.join('src', 'crypto', 'test'), AllFiles) +
-      FindHeaderFiles(os.path.join('src', 'ssl', 'test'), NoTestRunnerFiles))
+      FindHeaderFiles(os.path.join(SRC_DIR, 'crypto', 'test'), AllFiles) +
+      FindHeaderFiles(os.path.join(SRC_DIR, 'ssl', 'test'), NoTestRunnerFiles))
 
   crypto_test_files = []
-  if EMBED_TEST_DATA:
-    # Generate crypto_test_data.cc
-    with open('crypto_test_data.cc', 'w+') as out:
-      subprocess.check_call(
-          ['go', 'run', 'util/embed_test_data.go'] + cmake['CRYPTO_TEST_DATA'],
-          cwd='src',
-          stdout=out)
-    crypto_test_files += ['crypto_test_data.cc']
-
-  crypto_test_files += FindCFiles(os.path.join('src', 'crypto'), OnlyTests)
+
+  # Generate crypto_test_data.cc
```

Comment:
We do this unconditionally now? I see that previously it was done only `if EMBED_TEST_DATA`

Reply:
Yes, this was a conditional before but unless specified, `EMBED_TEST_DATA` defaulted to `True`. This would only be false for Bazel or GN build systems, which we removed support for since we do not test against those.